src/core/service.c

```diff
@@ -381,7 +381,7 @@ static int service_add_fd_store(Service *s, int fd, const char *name) {
                 if (r > 0) {
                         /* Already included */
                         safe_close(fd);
-                        return 1;
+                        return 0;
```

Comment:
hmm, this looks wrong. With this change the caller cannot know whether the fd is still in its possession or not. Previously a return value of "0" meant: "I didn't add it, it remains in your possession", and "1"meant: "I took possession of it, you lost possession of it". But with your change 0 can mean both now...